scalar/src/cuda/trid_iterative_mpi.hpp

```diff
@@ -813,7 +830,7 @@ __global__ void trid_linear_backward_pass_unaligned(
       int ind_floor = ((ind + offset) / align<REAL>)*align<REAL> - offset;
       int sys_off   = ind - ind_floor;
       int end_remainder =
-          sys_size / vec_length<REAL> * vec_length<REAL> - sys_off;
+          ((sys_size - offset) / vec_length<REAL>) * vec_length<REAL> - sys_off;
```

Comment:
I see what you mean, I think I originally misunderstood this. The 'fix' I had previously was passing the OPS tests because it was causing another part of the backward pass code to be skipped instead of the 'fix' actually being correct. 

So I've removed what I did with - offset and instead changed n + i == sys_size - 1 to n + i - sys_off == sys_size - 1 in the section of the code that does the unaligned loads and stores. This passes the OPS and Tridsolver tests, and I think solves the original issue I was having. Does this look right to you?

Reply:
The new updates is also ok. our OPS tests are working as before.

Reply:
Yes, this look good to me.